sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRequest.java

```diff
@@ -278,4 +289,102 @@ public HttpRequest copy() {
+        if (body == null) {
+            return null;
+        }
+
+        if (bufferingMode == HttpRequestBodyBufferingMode.NEVER) {
+            return body;
+        }
+
+        BinaryDataContent content = BinaryDataHelper.getContent(body);
+
+        if (content instanceof ByteArrayContent
+            || content instanceof SerializableContent
+            || content instanceof StringContent
+            || content instanceof FileContent) {
+            return body;
+        } else if (content instanceof InputStreamContent) {
+            InputStream inputStream = content.toStream();
+            Long contentLength = getContentLength();
+            if (contentLength != null && contentLength < Integer.MAX_VALUE) {
+                if (inputStream.markSupported() && bufferingMode == HttpRequestBodyBufferingMode.SHALLOW) {
+                    if (requestBodyCopied) {
+                        try {
+                            inputStream.reset();
+                        } catch (IOException e) {
+                            throw LOGGER.logExceptionAsError(
+                                new UncheckedIOException("Unable to reset request body", e));
+                        }
+                    }
+                    inputStream.mark(contentLength.intValue());
+                    return body;
+                } else {
+                    // If stream is not seekable buffer body and set it in both derived and original request.
+                    byte[] bufferedContent = content.toBytes();
+                    BinaryData bufferedBinaryData = BinaryData.fromBytes(bufferedContent);
+                    this.setBody(bufferedBinaryData);
+                    return bufferedBinaryData;
+                }
+            } else {
+                throw LOGGER.logExceptionAsError(new IllegalStateException("Non retryable request body"));
+            }
+        } else {
+            /*
+             Clone the original request to ensure that each try starts with the original (unmutated) request. We cannot
+             simply call httpRequest.buffer() because although the body will start emitting from the beginning of the
+             stream, the buffers that were emitted will have already been consumed (their position set to their limit),
+             so it is not a true reset. By adding the map function, we ensure that anything which consumes the
+             ByteBuffers downstream will only actually consume a duplicate so the original is preserved. This only
+             duplicates the ByteBuffer object, not the underlying data.
+             */
+            if (bufferingMode == HttpRequestBodyBufferingMode.SHALLOW) {
+                Flux<ByteBuffer> bufferedBody = body.toFluxByteBuffer().map(ByteBuffer::duplicate);
+                return BinaryDataHelper.createBinaryData(new FluxByteBufferContent(bufferedBody, body.getLength()));
+            } else {
+                byte[] bufferedContent = content.toBytes();
```

Comment:
Same comment as above, and maybe this requires a `List<byte[]>` `BinaryDataContent` type such as `SequenceByteArrayContent`